src/cache/memory.rs

```diff
@@ -0,0 +1,46 @@
+use std::{borrow::BorrowMut, collections::HashMap, time::Duration};
+
+use tokio::sync::Mutex;
+
+use super::Cache;
+use crate::twin::Twin;
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use serde_json::Value;
+
+pub struct MemCache<V> {
+    mem: Mutex<HashMap<String, V>>,
```

Comment:
It's better (in this specific use case) to use the `tokio::sync::RwLock` instead of the `Mutex` because RwLock can allow you to lock in `exclusive` (write) mode which means one routing can have mut reference of the object. OR `shared` (read) mode which allows multiple instances  to hold a read only reference to the `mem` object.

In other words, this will make it possible that multiple routines will be able to read from the map at the same time. But only on updating the map, only one routine can do this at a time.

This way `get` operations will not block uncessary. because 2 routines (or more) can do get at the same time with no blocking.